src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -554,6 +555,34 @@ public static class HwIntrinsics
             return Avx.Add(Avx.Multiply(vm0, vm1), va);
         }
 
+        /// <summary>
+        /// Performs a multiplication and an addition of the <see cref="Vector128{Single}"/>.
+        /// TODO: Fix. The arguments are in a different order to the FMA intrinsic.
+        /// </summary>
+        /// <remarks>ret = (vm0 * vm1) + va</remarks>
+        /// <param name="va">The vector to add to the intermediate result.</param>
+        /// <param name="vm0">The first vector to multiply.</param>
+        /// <param name="vm1">The second vector to multiply.</param>
+        /// <returns>The <see cref="Vector256{T}"/>.</returns>
+        [MethodImpl(InliningOptions.AlwaysInline)]
+        public static Vector128<float> MultiplyAdd(
+            in Vector128<float> va,
+            in Vector128<float> vm0,
+            in Vector128<float> vm1)
```

Comment:
```suggestion
            Vector128<float> va,
            Vector128<float> vm0,
            Vector128<float> vm1)
```
The `in` modifier isn't needed here. `in` is a readonly-reference, thus the address of the arguments must be exposed (i.e. passed via stack and not via registers). Here the JIT will inline the method, so that "address exposure" won't happen. To make JIT's job easier just don't specify `in` here.

Reply:
Yeah, that one's on me.